torch/optim/swa_utils.py

```diff
@@ -98,7 +98,14 @@ def avg_fn(averaged_model_parameter, model_parameter, num_averaged):
                 return averaged_model_parameter + \
                     (model_parameter - averaged_model_parameter) / (num_averaged + 1)
         self.avg_fn = avg_fn
-        self.use_state_dict = mode == 'state_dict'
+        modes = {
+            'parameters': False,
+            'state_dict': True,
+        }
+        try:
+            self.use_state_dict = modes[mode]
```

Comment:
This is a valid pattern but I'm not sure if it's common across Core. I'll leave @albanD to give input of whether we can keep this or use if/elif/else there.

Reply:
We don't usually do this indeed.
One thing you can do is either:
```python
if mode not in modes:
  raise ...
```
or 
```
if mode == "foo""
   # do smt
elif mode == "bar":
  # do smt else
else:
  raise ...
```

In this case, I feel like the first one is going to be the simplest here:
Just add a check and keep the assignment as is.

Reply:
What is the benefit of having if..elif..else here? With a dictionary, it's easier to add more options later with minimal modification.

Reply:
Also, both options are same. Even if you do 
```
if mode not in modes:
  raise ...
```
you would still need to check for mode using if..else.

Reply:
The catching blanket error and re-throwing something else is not a great pattern for anything non-trivial (I agree that the current code is, but people might add more logic in there in the future). It might hide some other errors that you didn't want to hide.

> you would still need to check for mode using if..else.

That would look like this
```python
if mode not in ['state_dict', 'parameters']:
  raise RuntimeError()
self.use_state_dict = mode == 'state_dict'
```
After the check it is fine to assume that you have only there two modes that are valid.

The dictionary version here is not really more extensible as you store it in a boolean and you already use the two available values.

Reply:
Updated.